### PR TITLE
Adding HTXS Stage 1.2 classification

### PR DESCRIPTION
**Pull request by adewit**
#### PR description:

This adds the HTXS Stage 1.2 classification. Stage 1.1, 1 and 0 also remain available. 

#### PR validation:

Tested with runRivetHTXS_cfg.py on various Higgs signal samples

#### if this PR is a backport please specify the original PR:

Not a backport

### CODE REVIEW

File: GeneratorInterface/RivetInterface/src/HiggsTemplateCrossSections.cc [hunk 1]
```diff
@@ -85,6 +85,14 @@ namespace Rivet {
       return false;
     }
 
+    /// @brief Return true if particle decays to charged leptons.
+    bool ChLeptonDecay(const Particle &p) {
+      for (auto child : p.children())
+        if (PID::isChLepton(child.pid()))
```
Comment (fabiocos) [inline GeneratorInterface/RivetInterface/src/HiggsTemplateCrossSections.cc]: @adewit please set parentheses for bot the ```for``` loop and the ```if``` block to make the code readable in an unambiguous way

Comment (adewit) [reply GeneratorInterface/RivetInterface/src/HiggsTemplateCrossSections.cc]: Done - also made the same change to a few other functions which didn't have parentheses there either

**Comment by adewit** [comment]
@SiewYan @efeyazgan @mkirsano @agrohsje @alberto-sanchez @qliphy This has been open for 5 days but apart from Fabio's technical point I don't see a review / trigger of code tests /... When do you think you might get to this? We'd like to have this in the upcoming nanoAOD production so if any further changes are needed it would be good to converge on those soon

**Comment by qliphy** [comment]
please test

**Comment by qliphy** [comment]
Adding also @intrepid42 @xjanssen

**Comment by adewit** [comment]
Hi, sorry for asking again but is anyone reviewing this? On what timescale can we expect a decision on whether anything needs to be changed?

**Comment by intrepid42** [comment]
Hi, it seems to run with Rivet 3 so it's fine for me from Rivet view.

I suppose the actual routine has been agreed upon and checked by the Higgs XS working group?

**Comment by adewit** [comment]
Yes indeed, the routine was provided by the XSWG

**Comment by qliphy** [comment]
+1

**Comment by fabiocos** [comment]
+1

@peruzzim FYI, I understand you need a follow-up

**Pull request merged**